dbtsl/api/shared/query_params.py

```diff
@@ -1,4 +1,30 @@
-from typing import List, TypedDict
+from dataclasses import dataclass
+from typing import List, Optional, TypedDict, Union
+
+from dbtsl.models.time import TimeGranularity
+
+
+@dataclass(frozen=True)
+class OrderByMetric:
+    """Spec for ordering by a metric."""
+
+    name: str
+    descending: bool = False
+
+
+@dataclass(frozen=True)
+class OrderByDimension:
```

Comment:
We also need Entity support. But since GQL mushes Dimensions and Entities together as `GroupBy` we should probably use `OrderByGroupBy`.

Reply:
I can create a class for that ans use it like we use OrderByDimension

Reply:
I added a few fixup commits to rename this to `OrderByGroupBy`.